hubgrep/constants.py

```diff
@@ -37,33 +40,61 @@ class FORM_ARGS:
-TOOLTIP_FILTER_EXCLUDE_ARCHIVED = "Remove all repositories which are archived from search results."
-TOOLTIP_FILTER_EXCLUDE_MIRROR = "Remove all repositories which are mirrors from search results."
-TOOLTIP_FILTER_EXCLUDE_EMPTY = "Remove all empty repositories from search results."
-TOOLTIP_TAG_FORK = "This repository is a fork."
-TOOLTIP_TAG_ARCHIVED = "This repository has been archived."
-TOOLTIP_TAG_MIRROR = "This repository is a mirror - visit repo page to find original."
-TOOLTIP_TAG_EMPTY = "This repository has no content, only meta-data."
+TOOLTIP_FILTER_CREATED_BEFORE = gettext("Only include repositories that were created before this date.")
+TOOLTIP_FILTER_CREATED_AFTER = gettext("Only include repositories that were created after this date.")
+TOOLTIP_FILTER_MODIFIED_SINCE = gettext(
+    "Only include repositories that were modified after this date. Note that modified does not necessarily mean via push.")
+TOOLTIP_FILTER_PUSHED_SINCE = gettext("Only include repositories that have commits made to it after this date.")
+TOOLTIP_FILTER_EXCLUDE_SOURCE = gettext("Remove all repositories from results which belong to a hosting service.")
+TOOLTIP_FILTER_EXCLUDE_FORK = gettext("Remove all repositories which are forks from search results.")
+TOOLTIP_FILTER_EXCLUDE_ARCHIVED = gettext("Remove all repositories which are archived from search results.")
+TOOLTIP_FILTER_EXCLUDE_MIRROR = gettext("Remove all repositories which are mirrors from search results.")
+TOOLTIP_FILTER_EXCLUDE_EMPTY = gettext("Remove all empty repositories from search results.")
+TOOLTIP_TAG_FORK = gettext("This repository is a fork.")
+TOOLTIP_TAG_ARCHIVED = gettext("This repository has been archived.")
+TOOLTIP_TAG_MIRROR = gettext("This repository is a mirror - visit repo page to find original.")
+TOOLTIP_TAG_EMPTY = gettext("This repository has no content, only meta-data.")
 
 MISSING_VALUE_DEFAULT = "—"
 
-
+SEARCH_HINTS = [
+    {
+        "prefix": gettext("exact match:"),
+        "highlight": "\"alice carrol\"",
+        "suffix": gettext("Get results containing exact matches.")
+    },
+    {
+        "prefix": gettext("search by field:"),
+        "highlight": "alice @username carrol",
+        "suffix": gettext("Only carrol is matched against username.")
+    },
+    {
+        "prefix": gettext("exclude a word:"),
+        "highlight": "alice !wonderland",
+        "suffix": gettext("Only match results containing alice, but not if it also contains wonderland.")
+    },
+    {
+        "prefix": gettext("alternative search:"),
+        "highlight": "alice | wonderland",
+        "suffix": gettext("Will match results containing either alice OR wonderland.")
+    },
+    {
+        "prefix": gettext("preferential search:"),
+        "highlight": "alice MAYBE wonderland",
+        "suffix": gettext("Will mainly search for alice, but boost matches containing wonderland.")
+    },
+]
```

Comment:
maybe we should split out the tooltips and search_hints into its own file for kinda "content" stuff - but for now its in the same place.